### PR TITLE
Add advantages & challenges

### PR DESCRIPTION
**Pull request by carlydotwebsite**
System to add advantages and challenges to the character sheet. The user can now select them from a list, and edit the rating and specification of an advantage in a modal. The system will also show helpful descriptive data about a given advantage or challenge, when it exists, so that users don't need to constantly refer to the rules website.

https://tree.taiga.io/project/stephmarx-ask-again-later-character-system/task/101?kanban-status=856838
https://tree.taiga.io/project/stephmarx-ask-again-later-character-system/task/102?kanban-status=856838
https://tree.taiga.io/project/stephmarx-ask-again-later-character-system/task/103?kanban-status=856838
https://tree.taiga.io/project/stephmarx-ask-again-later-character-system/task/104?kanban-status=856838
https://tree.taiga.io/project/stephmarx-ask-again-later-character-system/task/89?kanban-status=856838
https://tree.taiga.io/project/stephmarx-ask-again-later-character-system/task/90?kanban-status=856838
https://tree.taiga.io/project/stephmarx-ask-again-later-character-system/task/91?kanban-status=856838
https://tree.taiga.io/project/stephmarx-ask-again-later-character-system/task/92?kanban-status=856838
https://tree.taiga.io/project/stephmarx-ask-again-later-character-system/task/93?kanban-status=856838
https://tree.taiga.io/project/stephmarx-ask-again-later-character-system/task/94?kanban-status=856838
https://tree.taiga.io/project/stephmarx-ask-again-later-character-system/task/95?kanban-status=856838
https://tree.taiga.io/project/stephmarx-ask-again-later-character-system/task/96?kanban-status=856838
https://tree.taiga.io/project/stephmarx-ask-again-later-character-system/task/97?kanban-status=856838

### CODE REVIEW

**Review by korey-enright — APPROVED** [review]
👍

File: app/controllers/storytellers/questionnaire_items_controller.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    end
+
+    def update
+  		@current_questions = QuestionnaireItem.all
+  		@questions = params[:questionnaire_item]
+  		@question_ids = []
+
+  		@questions.each do |question|
+  			if question[:question].present? #skip empties
+  				puts question.inspect
```
Comment (sublimemarch) [inline app/controllers/storytellers/questionnaire_items_controller.rb]: Debugging statement

File: app/views/characters/_form.slim [hunk 2]
```diff
@@ -108,3 +146,109 @@ javascript:
...
+    var advantageId = $selected.val();
+    var name = $selected.text().replace(/\s\[[0-9\,]+\]/, "");
+    var lowestRating;
+    if (ratings.length > 1) {
+      lowestRating = ratings.split(",")[0];
+    } else {
+      lowestRating = ratings;
+    }
+
+    console.log(specification);
```
Comment (sublimemarch) [inline app/views/characters/_form.slim]: debugging statement

File: app/views/characters/_form.slim [hunk 3]
```diff
@@ -108,3 +146,109 @@ javascript:
...
+      success: function(data) {
+        $modal.find('#modal-description').html(data.description);
+        $modal.find('.header span').text(data.name);
+        // hide all the stuff that's irrelevant
+        if (!data.requires_specification) {
+          $modal.find('.specification').hide();
+        } else {
+          $modal.find('.specification').show();
+        }
+        console.log(data.allowed_ratings.length > 1);
```
Comment (sublimemarch) [inline app/views/characters/_form.slim]: debugging statement

File: app/views/characters/_form.slim [hunk 4]
```diff
@@ -108,3 +146,109 @@ javascript:
...
+    $('.overlay').fadeIn();
+  });
+
+  $('#save-advantage').on('click', function(e) {
+    var $modal = $('#advantages-overlay .modal');
+    var specification = $modal.find('#modal-specification').val();
+    var rating = $modal.find('#modal-rating').val();
+    var advantageToUpdate = $modal.find('#modal-id').val();
+    var $advantage = $('#advantages-list li:nth-child('+(parseInt(advantageToUpdate)+1)+')');
+    console.log($advantage);
```
Comment (sublimemarch) [inline app/views/characters/_form.slim]: debugging statement

File: app/views/storytellers/questionnaire_items/index.slim [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+		li
+			= label_tag :question
+			= text_field_tag "questionnaire_item[][question]"
+			= hidden_field_tag "questionnaire_item[][id]"
+			= hidden_field_tag "questionnaire_item[][order]", "", class: 'order-field'
+
+  = button_tag "Add Question", type: 'button', id: 'add', class: 'button'
+  = button_tag "Save", type: 'submit', class: 'button'
+
+javascript:
```
Comment (sublimemarch) [inline app/views/storytellers/questionnaire_items/index.slim]: Let's extract this into an appropriate js file

**Review by sublimemarch — CHANGES_REQUESTED** [review]

File: app/views/storytellers/questionnaire_items/index.slim (hunk 5 above)
Comment (carlydotwebsite) [reply app/views/storytellers/questionnaire_items/index.slim]: done in another branch

File: app/views/characters/_form.slim (hunk 4 above)
Comment (carlydotwebsite) [reply app/views/characters/_form.slim]: removed all these console.log and debug statements

**Pull request merged**